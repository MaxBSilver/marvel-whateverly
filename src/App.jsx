import React, { Component } from 'react';
import './App.css';
import Banner from './components/Banner';
import Header from './layout/Header'

class App extends Component {
  constructor(){
    super()
      this.state = []
  }
  componentDidMount() {
    //todo: fetch the api here
  }
  render() {
    return (
      <div className="App">
        <Header />
        <Banner props={this.props}/>
      </div>
    );
  }
}

export default App;
