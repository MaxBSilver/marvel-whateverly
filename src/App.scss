body {
  margin: 10px;
}

h2 {
  font-family: $fancyfont;
  font-size: 2rem;
}
