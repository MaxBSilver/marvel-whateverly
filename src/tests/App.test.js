import React from 'react';
import ReactDOM from 'react-dom';
import App from '../App';
import { shallow } from 'enzyme';


describe('App', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow (
      <App />
    )
  });

  it('should have a proper default state', () => {
    expect(wrapper.state()).toEqual( {movies: [], comics: [], combined: [], rendered: []} )
  });

  it('should match the snapshot with all the data passed in', () => {
    expect(wrapper).toMatchSnapshot();
  })

})
