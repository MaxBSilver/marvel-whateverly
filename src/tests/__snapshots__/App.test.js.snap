// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot with all the data passed in 1`] = `
<div
  className="App"
>
  <Banner />
  <Header
    combined={Array []}
    comics={Array []}
    movies={Array []}
    rendered={Array []}
    storeRendered={[Function]}
  />
  <CardContainer
    rendered={Array []}
  />
</div>
`;
